client/my-sites/site-settings/redirect-to/redirect-to.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = ( WrappedComponent, redirectUrl ) => {
+	class RedirectNonJetpack extends Component {
+		static propTypes = {
+			redirectUrl: PropTypes.string,
+			// Connected props
+			siteIsAtomic: PropTypes.bool,
+			siteIsJetpack: PropTypes.bool,
+			siteSlug: PropTypes.string
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		verifySiteIsJetpackOrAtomic() {
```

Comment:
The method name is actually wrong now. The method will correctly redirect WP.com and Atomic sites as expected (and skip pure Jetpack sites), but in that case the name should rather be something like `verifySiteIsJetpackAndNotAtomic` (I'm open to any shorter alternatives, too, but if we want to be expressive with the method name, we'll have to keep the longer one).

Reply:
My proposition would be to describe the _behavior_ rather than the _implementation_. It's pretty trivial to look at the code and see what is actually being tested if that's relevant for you. Additionally, this method doesn't just verify, but actually _does_ the redirect if it's needed. Something like `redirectIfNecessary` or `maybeRedirect` perhaps?

Reply:
Re `Atomic`: my bad ("typo"), `NonAtomic` it is obviously. Where is the *facepalm* emoji when one needs it ? :)

I like the idea of renaming acc. to the behavior. Thanks!